crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -676,37 +655,49 @@ pub enum CumulocityConverterBuildError {
     OperationLogsError(#[from] OperationLogsError),
 }
 
-#[async_trait]
-impl Converter for CumulocityConverter {
-    type Error = ConversionError;
-
+impl CumulocityConverter {
     fn get_mapper_config(&self) -> &MapperConfig {
         &self.mapper_config
     }
 
-    async fn try_convert(&mut self, message: &Message) -> Result<Vec<Message>, ConversionError> {
-        let mut registration_messages = vec![];
+    pub async fn try_convert(
+        &mut self,
+        message: &Message,
+    ) -> Result<Vec<Message>, ConversionError> {
+        match self.mqtt_schema.entity_channel_of(&message.topic) {
+            Ok((source, channel)) => self.try_convert_te_topics(source, channel, message).await,
+            Err(_) => self.try_convert_tedge_topics(message).await,
+        }
```

Comment:
To ease the transition there is now two separate methods, each used for a specific topic schema.

The end goal is to get rid of `try_convert_tedge_topics`.